engine/class_modules/paladin/sc_paladin_holy.cpp

```diff
@@ -219,23 +218,22 @@ struct holy_prism_t : public paladin_spell_t
 
 struct holy_shock_damage_t : public paladin_spell_t
 {
-  double crit_chance_multiplier;
+  double crit_chance_boost;
 
   holy_shock_damage_t( paladin_t* p ) : paladin_spell_t( "holy_shock_damage", p, p->find_spell( 25912 ) )
   {
     background = may_crit = true;
     trigger_gcd           = 0_ms;
-
-    // this grabs the 100% base crit bonus from 20473
-    crit_chance_multiplier = p->find_class_spell( "Holy Shock" )->effectN( 1 ).base_value() / 10.0;
+    // this grabs the 30% base crit bonus from 272906
+    crit_chance_boost = p->spec.holy_shock_2->effectN( 1 ).percent();
```

Comment:
Since holy_shock_2's data and effects are already stored, you don't really have to store it here and can just call it from the composite_crit_chance() overrides. This is only necessary for values that you access with find_spell() and similar functions. (That said your code is perfectly fine, just giving you a heads up if you want to be lazy later)

Reply:
I like this since it makes me feel like my code is more optimized (as in what I have)